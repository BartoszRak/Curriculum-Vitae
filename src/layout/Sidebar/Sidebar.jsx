import React from 'react'
import PropTypes from 'prop-types'
import { withStyles, Paper } from '@material-ui/core'
import style from './Sidebar.style'

export function Sidebar({ classes }) {
  return (
    <Paper className={classes.root} square>
      Sidebar here
    </Paper>
  )
}

Sidebar.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Sidebar)
