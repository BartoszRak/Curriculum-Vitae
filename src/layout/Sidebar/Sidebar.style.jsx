export default ({ spacing }) => ({
  root: {
    background: 'none',
    boxShadow: 'none',
    height: '100%',
    left: 0,
    padding: spacing.unit * 2,
    position: 'fixed',
    top: 0,
    width: '15vw',
  }
})
