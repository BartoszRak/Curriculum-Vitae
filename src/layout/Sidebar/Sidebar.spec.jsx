import React from 'react'
import { shallow } from 'enzyme'

import { Sidebar } from './Sidebar'

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Sidebar {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Sidebar', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
