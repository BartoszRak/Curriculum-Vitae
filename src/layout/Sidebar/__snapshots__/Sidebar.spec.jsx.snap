// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Sidebar should render as expected 1`] = `
<PaperMock
  square={true}
>
  Sidebar here
</PaperMock>
`;
